data_layout.go

```diff
@@ -0,0 +1,421 @@
+package irutil
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// DataLayout is a structural representation of the datalayout
+// string from https://llvm.org/docs/LangRef.html#data-layout
+type DataLayout struct {
+	isBigEndian               bool                                   // default - True
+	naturalStackAlignment     uint64                                 // bits in multiple of 8, default: 0
+	programMemoryAddressSpace uint64                                 // default - 0
+	globalVarAddressSpace     uint64                                 // default - 0
+	allocaAddressSpace        uint64                                 // default - 0
+	ptrSizeAlign              map[uint64]*PointerSizeAlignment       // Key is addressspace
+	intSizeAlign              map[uint64]*IntegerSizeAlignment       // Key is size
+	vecSizeAlign              map[uint64]*VectorSizeAlignment        // Key is size
+	floatSizeAlign            map[uint64]*FloatingPointSizeAlignment // Key is size
```

Comment:
Does there have any reasons we need to use a pointer to the structure instead of the structure itself?

Reply:
I am not sure I understand the question correctly. 
We need instances of those structures and pointer is the way to represent it right ?

Reply:
For example

map[uint64]VectorSizeAlignment

instead of 

map[uint64]*VectorSizeAlignment

Reply:
> For example `map[uint64]VectorSizeAlignment` instead of `map[uint64]*VectorSizeAlignment`
> Go does not require structure with pointer

Doing this would mean, they are passed by values when they are given as arguments or returned from functions, thereby changes made to them not getting reflected if not re assigned properly. Whereas, pointers would reflect any changes made to it no matter how an instance is used. I think its better this way.

Reply:
Not actually, because you can use & to get a reference whenever you want

Reply:
That can be done, but that is one more thing to include in review to make sure we haven't forgotten to add & wherever required. 

Anyway, I checked for llir/llvm repo and the convention there is use of pointers in case of structs https://github.com/llir/llvm/blob/9e62c17a9f75d127c4010b0d242912413c9d73cc/ir/func.go#L19 so I think this follows the existing pattern.